src/arch/x86/analyzex86.rs

```diff
@@ -454,27 +423,34 @@ pub struct MemoryManager<'a>
     pub list: Vec<MemoryBounds<'a>>,
 }
 
-fn read_int<T: Num + BitOr<Output=T> + NumCast + Copy>(
-     _byte: &mut T,
+pub fn read_int<T: Num + NumCast + Copy + 'static>(
      _offset: usize,
      _address: usize,
-     _size: usize,
-     _binary: &[u8])-> bool
+     _binary: &[u8]) -> Option<T>
+    where u64: AsPrimitive<T>
 {
-    let length = ::std::mem::size_of::<T>();
+    if let Some(start) = _offset.checked_sub(_address) {
```

Comment:
`read_int` just checks that the whole value to read is in range, which means that bounds checks can be deferred at call sites - this introduces overhead in a few uses where the read is assumed to be valid, but also removes some duplicate checks elsewhere. This is also why I want to figure out what `index_check` is doing over in `stack_read`; if it's doing a bounds check it's redundant with logic here, I just didn't figure out how to shuffle the values around right